src/secret_sharing/replicated/malicious/additive_share.rs

```diff
@@ -23,16 +23,37 @@ use std::{
 use typenum::Unsigned;
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct AdditiveShare<V: SharedValue> {
+pub struct AdditiveShare<V: SharedValue + ExtendableField> {
     x: SemiHonestAdditiveShare<V>,
-    rx: SemiHonestAdditiveShare<V>,
+    rx: SemiHonestAdditiveShare<V::LargeFieldType>,
 }
 
-impl<V: SharedValue> SecretSharing<V> for AdditiveShare<V> {
+pub trait ExtendableField {
+    type LargeFieldType: Field;
+    fn get_induced_value(&self) -> Self::LargeFieldType;
```

Comment:
With suggestion 1 above, you should maybe call this `to_induced()`:

```suggestion
    fn to_induced(&self) -> Self::InducedField;
```

or with suggestion 2:

```suggestion
    fn to_extended(&self) -> Self::ExtendedField;
```

Brevity, consistency and all that.  The paper tends to use "induced", but it seems like you might prefer "extended".  I don't quite know which is bette.r

Reply:
OK, I'll go with "to_extended()"